packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -189,9 +201,219 @@ const renderQueryEditorHeaderLabel = (
+            {!editorStore.isLoadingQueryInProgress && (
+              <>
+                {editorStore.queries.length > 0 && (
+                  <>
+                    <table className="table">
+                      <thead style={{ position: 'sticky', top: -1 }}>
+                        <tr>
+                          {queryLoaderColumns.map((column, idx) => (
+                            <th
+                              key={column}
+                              style={{ fontWeight: 'bold' }}
+                              className="table__cell--left"
+                            >
+                              {column}
+                            </th>
+                          ))}
+                        </tr>
+                      </thead>
+                      <tbody>
+                        {(editorStore.isMineOnly
+                          ? editorStore.queries.filter(
+                              (q) => q.isCurrentUserQuery,
+                            )
+                          : editorStore.queries
+                        ).map((q) => (
+                          <tr
+                            key={q.id}
+                            className={clsx(
+                              'query-editor__query-loader__body__table__row',
+                              {
+                                'query-editor__query-loader__body__table__row--selected':
+                                  selectedQueryID === q.id,
+                              },
+                            )}
+                            onClick={(event) => setSelectedQueryID(q.id)}
+                          >
+                            {buildQueryTableRow(q).map((value, columnIdx) => (
+                              // eslint-disable-next-line react/no-array-index-key
+                              <td key={columnIdx} className="table__cell--left">
+                                {value}
+                              </td>
+                            ))}
+                          </tr>
+                        ))}
+                      </tbody>
+                    </table>
+                    <div className="query-editor__query-loader__body__warning">
+                      Show relevant queries by searching names
+                    </div>
+                  </>
+                )}
+                {editorStore.queries.length === 0 && (
+                  <BlankPanelContent>No Query to Preview</BlankPanelContent>
+                )}
+              </>
+            )}
+            {editorStore.isLoadingQueryInProgress && (
+              <>
+                <PanelLoadingIndicator
+                  isLoading={editorStore.isLoadingQueryInProgress}
+                />
+                <BlankPanelContent>loading queries...</BlankPanelContent>
+              </>
+            )}
+          </div>
+          <div className="search-modal__actions">
+            <button className="btn btn--dark" onClick={loadSelectedQuery}>
+              Load
```

Comment:
-> `Load Query`

Also while we're here, I think we should create an issue to handle these kinds of flow, user could lose their progress on the query here and we should block that, the same way we block in `Studio`.